src/tests/suite/daos_aggregate_ec.c

```diff
@@ -0,0 +1,407 @@
+		assert_int_equal(rc, 0);
+		rc = dc_task_schedule(task, true);
+		assert_int_equal(rc, 0);
+		/* verify no remaining replicas on parity target */
+		assert_int_equal(ctx.fetch_iom.iom_nr_out, 0);
+		task = NULL;
+		memset(&ctx.fetch_iom, 0, sizeof(daos_iom_t));
+		ctx.fetch_iom.iom_flags = DAOS_IOMF_DETAIL;
+		ctx.fetch_iod.iod_recxs[0].rx_idx = (i * len) |
+						     PARITY_INDICATOR;
+		ctx.fetch_iod.iod_recxs[0].rx_nr = len;
+		ctx.iom_recx.rx_nr = len;
+		rc = dc_obj_fetch_task_create(ctx.oh, DAOS_TX_NONE, 0,
+					      &ctx.dkey, 1, DIOF_TO_SPEC_SHARD,
+					      &ctx.fetch_iod, &ctx.fetch_sgl,
+					      &ctx.fetch_iom, &shard, NULL,
+					      NULL, &task);
+		assert_int_equal(rc, 0);
+		rc = dc_task_schedule(task, true);
+		assert_int_equal(rc, 0);
+		/* verify parity now exists on parity target */
+		assert_int_equal(ctx.fetch_iom.iom_nr_out, 1);
+		task = NULL;
+	}
+	for (i = 0; i < 64; i++) {
+		ec_setup_single_recx_data(&ctx, EC_SPECIFIED, i * (2 * len),
+					  2 * len, 2);
+		buf = ctx.update_sgl.sg_iovs[0].iov_buf;
+		for (j = 0; j < k; j++)
+			data[j] = &buf[j * len];
+		ec_encode_data(len, k, p, codec->ec_gftbls, data, parity);
+		ctx.fetch_iom.iom_flags = DAOS_IOMF_DETAIL;
+		rc = dc_obj_fetch_task_create(ctx.oh, DAOS_TX_NONE, 0,
+					      &ctx.dkey, 1, DIOF_TO_SPEC_SHARD,
+					      &ctx.fetch_iod, &ctx.fetch_sgl,
+					      &ctx.fetch_iom, &shard, NULL,
+					      NULL, &task);
+		assert_int_equal(rc, 0);
+		rc = dc_task_schedule(task, true);
+		assert_int_equal(rc, 0);
+		/* verify no remaining replicas on parity target */
+		assert_int_equal(ctx.fetch_iom.iom_nr_out, 0);
+		task = NULL;
+		memset(&ctx.fetch_iom, 0, sizeof(daos_iom_t));
+		ctx.fetch_iom.iom_flags = DAOS_IOMF_DETAIL;
+		ctx.fetch_iod.iod_recxs[0].rx_idx = (i * len) |
+						     PARITY_INDICATOR;
+		ctx.fetch_iod.iod_recxs[0].rx_nr = len;
+		ctx.iom_recx.rx_nr = len;
+		rc = dc_obj_fetch_task_create(ctx.oh, DAOS_TX_NONE, 0,
+					      &ctx.dkey, 1, DIOF_TO_SPEC_SHARD,
+					      &ctx.fetch_iod, &ctx.fetch_sgl,
+					      &ctx.fetch_iom, &shard, NULL,
+					      NULL, &task);
+		assert_int_equal(rc, 0);
+		rc = dc_task_schedule(task, true);
+		assert_int_equal(rc, 0);
+		/* verify parity now exists on parity target */
+		assert_int_equal(ctx.fetch_iom.iom_nr_out, 1);
+		task = NULL;
+	}
+	ec_cleanup_data(&ctx);
+	ec_cleanup_cont_obj(&ctx);
+}
+
+
+static int
+ec_setup(void **statep)
+{
+	return test_setup(statep, SETUP_POOL_CONNECT, true, DEFAULT_POOL_SIZE,
+			  NULL);
+}
+
+
+static const struct CMUnitTest ec_agg_tests[] = {
+	{"DAOS_ECAG00: test_filled_stripe", test_filled_stripe,
+	  incremental_fill, test_case_teardown},
+};
+
+
+int run_daos_aggregation_ec_test(int rank, int size, int *sub_tests,
+				 int sub_tests_size)
+{
+
+	int rc = 0;
+
+	if (rank != 0) {
+		MPI_Barrier(MPI_COMM_WORLD);
+		return rc;
+	}
+	rc = cmocka_run_group_tests_name("DAOS EC AGGREGATION TESTS",
+					 ec_agg_tests, ec_setup, test_teardown);
```

Comment:
I think bunch of unit test cases need be introduced to verify if EC aggregation handles replica, parity combinations correctly. It would be similar to what we did for VOS aggregation, generating all kind of combinations of replicas and parity for single/multiple stripes, and verify if the left replica count & content are expected.

Let's improve the test cases post milestone.